webapplication/user/admin.py

```diff
@@ -48,3 +49,17 @@ class TransactionModel(admin.ModelAdmin):
             'fields': ('amount', 'user', 'request', 'comment', 'completed')
         }),
     )
+
+    def top_up_user_balance(self, request, obj):
+        with transaction.atomic():
+            is_updated = obj.user.top_up_balance(obj.amount)
+            if is_updated:
+                obj.completed = True
+                obj.save(update_fields=("completed",))
+            else:
+                messages.error(request, _("Error caused in user balance top up!"))
+
+    def save_model(self, request, obj, form, change):
+        super().save_model(request, obj, form, change)
+        if not change:
+            self.top_up_user_balance(request, obj)
```

Comment:
Better in user admin add ability to change the balance. If balance is changed the corresponding transaction is added. Note that amount can be negative and positive.